packages/contracts/src/registry/SpaceRegistry.sol

```diff
@@ -0,0 +1,131 @@
+/// @dev This contract serves as the entry point for creating new spaces by deploying DAO contracts.
+///      Each space has a unique ID that maps to a DAO contract address. Users can designate one space
+///      as their "home space". To set an existing space as home, users must request it and the 
+///      space's DAO must accept. Spaces can migrate to new DAO contracts while keeping their ID.
+contract SpaceRegistry is OwnableUpgradeable, UUPSUpgradeable,ISpaceRegistry {
+
+    /// @notice The DAOFactory contract used to deploy new DAO instances
+    DAOFactory public daoFactory;
+    
+    /// @notice Maps each unique space ID to its current DAO contract address
+    mapping(bytes16 => address) public daoAddressBySpaceId;
+    
+    /// @notice Reverse mapping: DAO address to its space ID
+    mapping(address => bytes16) public spacesByDAOAddress;
+    
+    /// @notice The home space ID for each user address (bytes16(0) if none)
+    mapping(address => bytes16) public homeSpaceByAddress;
+    
+    /// @notice Pending home space requests: user address to requested space ID
+    /// @dev When a user wants to set an existing space as home, the space ID is stored here
+    ///      until the space's DAO accepts. Using space IDs (not DAO addresses) ensures 
+    ///      requests remain valid if the space migrates to a new DAO contract.
+    mapping(address => bytes16) public pendingHomeSpaceId;
+
+    /// @notice Initializes the SpaceRegistry contract
+    /// @param _owner The address that will own this registry contract
+    /// @param _daoFactory The address of the DAOFactory contract used to deploy spaces
+    function initialize(address _owner, address _daoFactory) external initializer {
+        if(_daoFactory == address(0)) revert SpaceRegistryInvalidZeroAddress();
+
+        __Ownable_init();
+        _transferOwnership(_owner);
+        daoFactory = DAOFactory(_daoFactory);
+
+        emit SpaceRegistryInitialized(_daoFactory, _owner);
+    }
+
+    /// @inheritdoc ISpaceRegistry
+    function createSpace(DAOFactory.DAOSettings calldata _daoSettings, DAOFactory.PluginSettings[] calldata _pluginSettings, bool _isHomeSpace) external override returns (DAO createdDao, bytes16 spaceId) {
+        createdDao = daoFactory.createDao(_daoSettings, _pluginSettings);
+        spaceId = generateSpaceId(address(createdDao));
+        daoAddressBySpaceId[spaceId] = address(createdDao);
+        spacesByDAOAddress[address(createdDao)] = spaceId;
+
+        if(_isHomeSpace) {
+            bytes16 previousHomeSpace = homeSpaceByAddress[msg.sender];
+            homeSpaceByAddress[msg.sender] = spaceId;
+            emit SpaceRegistryHomeSpaceSet(msg.sender, previousHomeSpace, spaceId);
+        }
+
+        emit SpaceRegistrySpaceCreated(spaceId, address(createdDao), msg.sender);
+    }
+
+    /// @inheritdoc ISpaceRegistry
+    function setHomeSpace(bytes16 _spaceId) external override {
```

Comment:
where can i find docs on what's expected to be a home space?
is it possible to 1 space be homespace of N daos?

Reply:
been reading "Space contracts and ids" and still it isn't clear to me, there are no user stories or features on why a `msg.sender` would set a `spaceId` as "home", i think it'd be interesting to go deep into it, given there are many flows (i.e. "the dao accepting the space as user's home", `SpaceRegistryHomeSpaceUpdatePending` event, ...) that depend on this feature 👀